### PR TITLE
zephyr: direct inclusion of the hooks file

### PR DESCRIPTION
**Pull request by nvlsianpu**
This patch removes CONFIG_BOOT_IMAGE_ACCESS_HOOKS_FILE option
which was used to point to hooks implementation file.
It is better that the project customization is up to add required
files instead. This gives more flexibility in provisioning these
sources (there may be multiple files).
This also simplifies Kconfig options.

In order to keep possibility for build-test the Hooks feature
hooks_sample.c will be included into the build if commandline
will specify -DTEST_BOOT_IMAGE_ACCESS_HOOKS=Y option.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Comment by nvlsianpu** [comment]
cc @tejlmand

**Review by tejlmand — COMMENTED** [review]
Thanks for this cleanup.

I believe the use of `TEST_BOOT_IMAGE_ACCESS_HOOKS` should have some comment in the CMake code, but as this is not blocking I'm giving +1

File: boot/bootutil/zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -17,27 +17,10 @@ zephyr_library_sources(
...
-  endif()
-
-  if(DEFINED HOOKS_FILE)
-      zephyr_library_sources(
-        ${HOOKS_FILE}
-      )
-  else()
-    message(STATUS "No hooks implementation file.")
-  endif()
+if(TEST_BOOT_IMAGE_ACCESS_HOOKS)
```
[the remark this answers is not in the document]
Comment (nvlsianpu) [reply boot/bootutil/zephyr/CMakeLists.txt:22]: Comment added.

**Review by de-nordic — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: boot/bootutil/zephyr/CMakeLists.txt [hunk 2]
```diff
@@ -17,27 +17,12 @@ zephyr_library_sources(
...
-  endif()
-
-  if(DEFINED HOOKS_FILE)
-      zephyr_library_sources(
-        ${HOOKS_FILE}
-      )
-  else()
-    message(STATUS "No hooks implementation file.")
-  endif()
+# Sensitivit to the TEST_BOOT_IMAGE_ACCESS_HOOKS define is implemented for
```
Comment (gustavonihei) [inline boot/bootutil/zephyr/CMakeLists.txt:20]:
```suggestion
# Sensitivity to the TEST_BOOT_IMAGE_ACCESS_HOOKS define is implemented for
```
nit: Typo in documentation

**Review by gustavonihei — APPROVED** [review]
LGTM, just found one minor nitpick.

**Pull request merged**